config/config.go

```diff
@@ -778,6 +784,10 @@ func (c *ScrapeConfig) SetDirectory(dir string) {
 // UnmarshalYAML implements the yaml.Unmarshaler interface.
 func (c *ScrapeConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	*c = DefaultScrapeConfig
+	// Unset default validation and escaping scheme values so they can inherit
+	// from the global config if they are blank in the file.
+	c.MetricNameValidationScheme = ""
+	c.MetricNameEscapingScheme = ""
```

Comment:
Why do we need to do this only for these config options?  We don't seem to reset other config options (e.g. ScrapeProtocols) that inherit from the global config.

Reply:
Hm well it used to be that there was not a static "default" for these values.  But now Validation Scheme always defaults to UTF-8, whereas before it might be "legacy" if the library global was set that way.  But Escaping Scheme does have a variable default -- underscores if the validation scheme is legacy, or allow-utf-8.

The lack of predictable default is the reason for the trickiness.

Reply:
Right... I guess my suggestion is that we don't set these in `DefaultScrapeConfig`.  It seems like `(c *ScrapeConfig) Validate` already handles fallbacks for these two fields to the global one.  You could add these fields to the comment at the top of DefaultScrapeConfig.  We can handle properly defaulting escaping scheme in Validate() below as well.

Reply:
this is what worries me: https://github.com/search?q=config.DefaultScrapeConfig&type=code 

a lot of calls to this code without validation

Reply:
We do that for a bunch of other fields, so that must be OK?